File: CreaObjetos/CreaObjetos/GeneraCodigo/generaDiseno.vb
Public Class generaDiseno
    Public neimespeis As String
    Public clace As Clase
    Dim tabIndex As Integer
    Public Function generaDiseno() As String
        Dim codigo As String
        codigo = " <Global.Microsoft.VisualBasic.CompilerServices.DesignerGenerated()> _" & vbCrLf
        codigo &= " Partial Class frm" & clace.NombreTabla & vbCrLf
        codigo &= "    Inherits System.Windows.Forms.Form " & vbCrLf


        codigo &= Me.creaDispose()
        codigo &= Me.creaInitializeComponent()
        codigo &= Me.creaFriends()
        codigo &= "End Class" & vbCrLf

        Return codigo
    End Function

    Public Function creaDispose()
        Dim code As New String("")
        code &= "'Form overrides dispose to clean up the component list." & vbCrLf
        code &= "<System.Diagnostics.DebuggerNonUserCode()> _" & vbCrLf
        code &= "Protected Overrides Sub Dispose(ByVal disposing As Boolean)" & vbCrLf
        code &= "Try" & vbCrLf
        code &= "If disposing AndAlso components IsNot Nothing Then" & vbCrLf
        code &= " components.Dispose()" & vbCrLf
        code &= "End If" & vbCrLf
        code &= "Finally" & vbCrLf
        code &= " MyBase.Dispose(disposing)" & vbCrLf
        code &= "End Try" & vbCrLf
        code &= "End Sub" & vbCrLf

        Return code

    End Function

    Public Function creaInitializeComponent()
        Dim code As New String("")
        code &= "   'Required by the Windows Form Designer" & vbCrLf
        code &= "   Private components As System.ComponentModel.IContainer" & vbCrLf

        code &= "   'NOTE: The following procedure is required by the Windows Form Designer" & vbCrLf
        code &= "   'It can be modified using the Windows Form Designer.  " & vbCrLf
        code &= "   'Do not modify it using the code editor." & vbCrLf
        code &= "   <System.Diagnostics.DebuggerStepThrough()> _" & vbCrLf
        code &= "   Private Sub InitializeComponent()" & vbCrLf
        code &= "       Me.pnlDatos = New System.Windows.Forms.Panel" & vbCrLf
        code &= "       Me.lblFalta = New System.Windows.Forms.Label" & vbCrLf
        code &= "       Me.CtlbuscaReg = New Sist" & Me.neimespeis & ".ctlbuscaReg" & vbCrLf
        code &= "       Me.CtlABCMaestro = New Sist" & Me.neimespeis & ".ctlABCMaestro" & vbCrLf

        For Each atr As Propiedad In Me.clace.Campos

            Dim tipoCtl As String = atr.tipoControl
            Dim nonCtl As String = atr.nombreControl
            code &= "Me.lbl" & atr.Nombre & " =New System.Windows.Forms.Label" & vbCrLf

            If atr.Nombre.Substring(atr.Nombre.Length - 3, 3).Equals("_id") _
                And Not Me.clace.llave.Equals(atr.Nombre) Then
                tipoCtl = "cmb"
                nonCtl = "ComboBox"
            End If


            code &= "Me." & tipoCtl & atr.Nombre & "= New System.Windows.Forms." & nonCtl & vbCrLf
        Next

        code &= "       Me.pnlDatos.SuspendLayout()" & vbCrLf
        code &= "       Me.SuspendLayout()" & vbCrLf
        code &= Me.creaPanelDatos()
        code &= Me.crealblFalta()
        code &= Me.creaBusqueda()
        code &= Me.creaMaestro()
        'Para iniciar los controles

        For Each atr As Propiedad In Me.clace.Campos
            code &= Me.creaControles(atr)
        Next

        code &= Me.creaForma() & vbCrLf
        code &= "End sub" & vbCrLf

        Return code
    End Function


    Public Function creaPanelDatos() As String
        Dim code As New String("")
        code &= "'" & vbCrLf
        code &= "'pnlDatos" & vbCrLf
        code &= "'" & vbCrLf
        code &= "Me.pnlDatos.BackColor = System.Drawing.Color.FromArgb("
        code &= "CType(CType(192, Byte), Integer), "
        code &= "CType(CType(255, Byte), Integer), "
        code &= "CType(CType(255, Byte), Integer))" & vbCrLf
        code &= "Me.pnlDatos.Controls.Add(Me.lblFalta)" & vbCrLf

        For Each atr As Propiedad In Me.clace.Campos
            Dim tipoCtl As String = atr.tipoControl
            
            If atr.Nombre.Substring(atr.Nombre.Length - 3, 3).Equals("_id") _
                And Not Me.clace.llave.Equals(atr.Nombre) Then
                tipoCtl = "cmb"
            End If
            code &= "Me.pnlDatos.Controls.Add(Me.lbl" & atr.Nombre & ")" & vbCrLf
            code &= "Me.pnlDatos.Controls.Add(Me." & tipoCtl & atr.Nombre & ")" & vbCrLf
        Next
        code &= "Me.pnlDatos.Location = New System.Drawing.Point(0, 50)" & vbCrLf
        code &= "Me.pnlDatos.Name = ""pnlDatos""" & vbCrLf
        code &= "Me.pnlDatos.Size = New System.Drawing.Size(300, " & (Me.clace.Campos.Count * 30) + 50 & ")" & vbCrLf
        code &= "Me.pnlDatos.TabIndex = 0" & vbCrLf
        Return code
    End Function

    Public Function crealblFalta() As String
        Dim code As New String("")

        code &= "'lblFalta" & vbCrLf
        code &= "'" & vbCrLf
        code &= "Me.lblFalta.AutoSize = True" & vbCrLf
        code &= "Me.lblFalta.Font = New System.Drawing.Font(""Microsoft Sans Serif"", 9.75!, "
        code &= "System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))" & vbCrLf
        code &= "Me.lblFalta.ForeColor = System.Drawing.Color.Red" & vbCrLf
        code &= "Me.lblFalta.Location = New System.Drawing.Point(100," & (Me.clace.Campos.Count - 1) * 30 & ")" & vbCrLf
        code &= "Me.lblFalta.Name = ""lblFalta"" " & vbCrLf
        code &= "Me.lblFalta.Size = New System.Drawing.Size(0, 16)" & vbCrLf
        code &= "Me.lblFalta.TabIndex = 5" & vbCrLf
        Return code

    End Function
    Public Function creaFriends()
        Dim code As New String("")
        code &= "Friend WithEvents CtlABCMaestro As Sist" & Me.neimespeis & ".ctlABCMaestro" & vbCrLf
        code &= "Friend WithEvents CtlbuscaReg As Sist" & Me.neimespeis & ".ctlbuscaReg" & vbCrLf
        code &= "Friend WithEvents pnlDatos As System.Windows.Forms.Panel" & vbCrLf
        code &= "Friend WithEvents lblFalta As System.Windows.Forms.Label" & vbCrLf
        For Each atr As Propiedad In Me.clace.Campos
            Dim tipoCtl As String = atr.tipoControl
            Dim nonCtl As String = atr.nombreControl
            code &= "Friend WithEvents lbl" & atr.Nombre & " As System.Windows.Forms.Label" & vbCrLf

            If atr.Nombre.Substring(atr.Nombre.Length - 3, 3).Equals("_id") _
                And Not Me.clace.llave.Equals(atr.Nombre) Then
                tipoCtl = "cmb"
                nonCtl = "ComboBox"
            End If
            code &= "Friend WithEvents " & tipoCtl & atr.Nombre & " As System.Windows.Forms." & nonCtl & vbCrLf
        Next

        Return code
    End Function
    Public Function creaControles(ByVal atr As Propiedad) As String
        Dim code As New String("")
        code &= Me.creaEtiqueta(atr)
        code &= Me.creaCapturador(atr)
        Return code
    End Function

    Public Function creaEtiqueta(ByVal atr As Propiedad) As String
        Dim code As New String("")
        code &= "'" & vbCrLf
        code &= "'lbl" & atr.titulo & vbCrLf
        code &= "'" & vbCrLf
        code &= "Me.lbl" & atr.titulo & ".AutoSize = True" & vbCrLf
        code &= "Me.lbl" & atr.titulo & ".Font = "
        code &= "New System.Drawing.Font(""Microsoft Sans Serif"", 9.75!, "
        code &= "System.Drawing.FontStyle.Bold, "
        code &= "System.Drawing.GraphicsUnit.Point, CType(0, Byte))" & vbCrLf
        code &= "Me.lbl" & atr.titulo & ".Location = New System.Drawing.Point(20, " & (25 * Me.tabIndex) + 5 & ")" & vbCrLf
        code &= "Me.lbl" & atr.titulo & ".Name = ""lbl" & atr.titulo & """" & vbCrLf
        code &= "Me.lbl" & atr.titulo & ".Size = New System.Drawing.Size(" & atr.Nombre.Length * 10 & ", 16)" & vbCrLf
        code &= "Me.lbl" & atr.titulo & ".TabIndex = 0" & vbCrLf
        code &= "Me.lbl" & atr.titulo & ".Text = """ & atr.titulo & """" & vbCrLf
        Return code
    End Function
    Public Function creaCapturador(ByVal atr As Propiedad) As String
        Dim code As New String("")
        Dim tipoCtl As String = atr.tipoControl

        If atr.Nombre.Substring(atr.Nombre.Length - 3, 3).Equals("_id") _
            And Not Me.clace.llave.Equals(atr.Nombre) Then
            tipoCtl = "cmb"

        End If
        code &= "'" & vbCrLf
        code &= "'" & tipoCtl & atr.titulo & vbCrLf
        code &= "'" & vbCrLf
        code &= "Me." & tipoCtl & atr.titulo & ".Enabled = False" & vbCrLf
        code &= "Me." & tipoCtl & atr.titulo & ".Location =  New System.Drawing.Point(150, " & (25 * Me.tabIndex) + 5 & ")" & vbCrLf
        code &= "Me." & tipoCtl & atr.titulo & ".Name = """ & tipoCtl & atr.titulo & """" & vbCrLf
        If atr.Nombre.Equals(Me.clace.llave) Then
            code &= "Me." & tipoCtl & atr.titulo & ".ReadOnly = True" & vbCrLf
        End If
        code &= "Me." & tipoCtl & atr.titulo & ".Size = New System.Drawing.Size(100, 20)" & vbCrLf
        code &= "Me." & tipoCtl & atr.titulo & ".TabIndex = " & Me.tabIndex & vbCrLf
        Me.tabIndex += 1
        Return code
    End Function
    Public Function creaForma() As String
        Dim code As New String("")
        code &= "'" & vbCrLf
        code &= "'frm" & Me.clace.NombreTabla & vbCrLf
        code &= "'" & vbCrLf
        code &= "Me.AutoScaleDimensions = New System.Drawing.SizeF(6.0!, 13.0!)" & vbCrLf
        code &= "Me.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font" & vbCrLf
        code &= "Me.ClientSize = New System.Drawing.Size(300," & (Me.clace.Campos.Count * 30) + 100 & ")" & vbCrLf
        code &= "Me.Controls.Add(Me.pnlDatos)" & vbCrLf
        code &= "Me.Controls.Add(Me.CtlbuscaReg)" & vbCrLf
        code &= "Me.Controls.Add(Me.CtlABCMaestro)" & vbCrLf
        code &= "Me.Name = ""frm" & Me.clace.NombreTabla & """" & vbCrLf
        code &= "Me.Text = """ & Me.clace.NombreTabla & """" & vbCrLf
        code &= "Me.pnlDatos.ResumeLayout(False)" & vbCrLf
        code &= "Me.pnlDatos.PerformLayout()" & vbCrLf
        code &= "Me.ResumeLayout(False)" & vbCrLf
        Return code
    End Function
    Public Function creaBusqueda() As String
        Dim code As New String("")
        code &= "'" & vbCrLf
        code &= "'CtlbuscaReg" & vbCrLf
        code &= "'" & vbCrLf
        code &= "Me.CtlbuscaReg.Location = New System.Drawing.Point(0, 0)" & vbCrLf
        code &= "Me.CtlbuscaReg.Name = ""CtlbuscaReg""" & vbCrLf
        code &= "Me.CtlbuscaReg.Size = New System.Drawing.Size(516, 25)" & vbCrLf
        code &= "Me.CtlbuscaReg.TabIndex = 1" & vbCrLf
        Return code
    End Function
    Public Function creaMaestro() As String
        Dim code As New String("")
        code &= "'" & vbCrLf
        code &= "'CtlABCMaestro" & vbCrLf
        code &= "'" & vbCrLf
        code &= "Me.CtlABCMaestro.Estado = 0" & vbCrLf
        code &= "Me.CtlABCMaestro.Location = New System.Drawing.Point(0, 27)" & vbCrLf
        code &= "Me.CtlABCMaestro.Name = ""CtlABCMaestro""" & vbCrLf
        code &= "Me.CtlABCMaestro.Size = New System.Drawing.Size(516, 25)" & vbCrLf
        code &= "Me.CtlABCMaestro.TabIndex = 0" & vbCrLf
        Return code
    End Function






End Class

File: CreaObjetos/CreaObjetos.ConvertedToC#/GeneraCodigo/generaDiseno.cs
using Microsoft.VisualBasic;

namespace WindowsApplication1
{
	public class creaDiseno
	{
		public string neimespeis;
		public Clase clace;
		int tabIndex;
		/// <summary>
		/// 
		/// </summary>
		/// <returns></returns>
		public string GeneraDiseno()
		{
			string codigo = null;
			codigo = " <Global.Microsoft.VisualBasic.CompilerServices.DesignerGenerated()> _" + Constants.vbCrLf;
			codigo += " Partial Class frm" + clace.NombreTabla + Constants.vbCrLf;
			codigo += "    Inherits System.Windows.Forms.Form " + Constants.vbCrLf;


			codigo += this.creaDispose();
			codigo += this.CreaInitializeComponent();
			codigo += this.creaFriends();
			codigo += "End Class" + Constants.vbCrLf;

			return codigo;
		}

		public object creaDispose()
		{
			string code = string.Empty;
			code += "'Form overrides dispose to clean up the component list." + Constants.vbCrLf;
			code += "<System.Diagnostics.DebuggerNonUserCode()> _" + Constants.vbCrLf;
			code += "Protected Overrides Sub Dispose(ByVal disposing As Boolean)" + Constants.vbCrLf;
			code += "Try" + Constants.vbCrLf;
			code += "If disposing AndAlso components IsNot Nothing Then" + Constants.vbCrLf;
			code += " components.Dispose()" + Constants.vbCrLf;
			code += "End If" + Constants.vbCrLf;
			code += "Finally" + Constants.vbCrLf;
			code += " MyBase.Dispose(disposing)" + Constants.vbCrLf;
			code += "End Try" + Constants.vbCrLf;
			code += "End Sub" + Constants.vbCrLf;

			return code;

		}

		/// <summary>
		/// 
		/// </summary>
		/// <returns></returns>
		public object CreaInitializeComponent()
		{
			string code = string.Empty;
			code += "   'Required by the Windows Form Designer" + Constants.vbCrLf;
			code += "   Private components As System.ComponentModel.IContainer" + Constants.vbCrLf;

			code += "   'NOTE: The following procedure is required by the Windows Form Designer" + Constants.vbCrLf;
			code += "   'It can be modified using the Windows Form Designer.  " + Constants.vbCrLf;
			code += "   'Do not modify it using the code editor." + Constants.vbCrLf;
			code += "   <System.Diagnostics.DebuggerStepThrough()> _" + Constants.vbCrLf;
			code += "   Private Sub InitializeComponent()" + Constants.vbCrLf;
			code += "       Me.pnlDatos = New System.Windows.Forms.Panel" + Constants.vbCrLf;
			code += "       Me.lblFalta = New System.Windows.Forms.Label" + Constants.vbCrLf;
			code += "       Me.CtlbuscaReg = New Sist" + this.neimespeis + ".ctlbuscaReg" + Constants.vbCrLf;
			code += "       Me.CtlABCMaestro = New Sist" + this.neimespeis + ".ctlABCMaestro" + Constants.vbCrLf;


			foreach (Propiedad atr in this.clace.Campos) {
				string tipoCtl = atr.tipoControl;
				string nonCtl = atr.nombreControl;
				code += "Me.lbl" + atr.Nombre + " =New System.Windows.Forms.Label" + Constants.vbCrLf;

				if (atr.Nombre.Substring(atr.Nombre.Length - 3, 3).Equals("_id") & !this.clace.llave.Equals(atr.Nombre)) {
					tipoCtl = "cmb";
					nonCtl = "ComboBox";
				}


				code += "Me." + tipoCtl + atr.Nombre + "= New System.Windows.Forms." + nonCtl + Constants.vbCrLf;
			}

			code += "       Me.pnlDatos.SuspendLayout()" + Constants.vbCrLf;
			code += "       Me.SuspendLayout()" + Constants.vbCrLf;
			code += this.creaPanelDatos();
			code += this.crealblFalta();
			code += this.creaBusqueda();
			code += this.creaMaestro();
			//Para iniciar los controles

			foreach (Propiedad atr in this.clace.Campos) {
				code += this.creaControles(atr);
			}

			code += this.creaForma() + Constants.vbCrLf;
			code += "End sub" + Constants.vbCrLf;

			return code;
		}


		public string creaPanelDatos()
		{
			string code = string.Empty;
			code += "'" + Constants.vbCrLf;
			code += "'pnlDatos" + Constants.vbCrLf;
			code += "'" + Constants.vbCrLf;
			code += "Me.pnlDatos.BackColor = System.Drawing.Color.FromArgb(";
			code += "CType(CType(192, Byte), Integer), ";
			code += "CType(CType(255, Byte), Integer), ";
			code += "CType(CType(255, Byte), Integer))" + Constants.vbCrLf;
			code += "Me.pnlDatos.Controls.Add(Me.lblFalta)" + Constants.vbCrLf;

			foreach (Propiedad atr in this.clace.Campos) {
				string tipoCtl = atr.tipoControl;

				if (atr.Nombre.Substring(atr.Nombre.Length - 3, 3).Equals("_id") & !this.clace.llave.Equals(atr.Nombre)) {
					tipoCtl = "cmb";
				}
				code += "Me.pnlDatos.Controls.Add(Me.lbl" + atr.Nombre + ")" + Constants.vbCrLf;
				code += "Me.pnlDatos.Controls.Add(Me." + tipoCtl + atr.Nombre + ")" + Constants.vbCrLf;
			}
			code += "Me.pnlDatos.Location = New System.Drawing.Point(0, 50)" + Constants.vbCrLf;
			code += "Me.pnlDatos.Name = \"pnlDatos\"" + Constants.vbCrLf;
			code += "Me.pnlDatos.Size = New System.Drawing.Size(300, " + (this.clace.Campos.Count * 30) + 50 + ")" + Constants.vbCrLf;
			code += "Me.pnlDatos.TabIndex = 0" + Constants.vbCrLf;
			return code;
		}

		public string crealblFalta()
		{
			string code = string.Empty;

			code += "'lblFalta" + Constants.vbCrLf;
			code += "'" + Constants.vbCrLf;
			code += "Me.lblFalta.AutoSize = True" + Constants.vbCrLf;
			code += "Me.lblFalta.Font = New System.Drawing.Font(\"Microsoft Sans Serif\", 9.75!, ";
			code += "System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))" + Constants.vbCrLf;
			code += "Me.lblFalta.ForeColor = System.Drawing.Color.Red" + Constants.vbCrLf;
			code += "Me.lblFalta.Location = New System.Drawing.Point(100," + (this.clace.Campos.Count - 1) * 30 + ")" + Constants.vbCrLf;
			code += "Me.lblFalta.Name = \"lblFalta\" " + Constants.vbCrLf;
			code += "Me.lblFalta.Size = New System.Drawing.Size(0, 16)" + Constants.vbCrLf;
			code += "Me.lblFalta.TabIndex = 5" + Constants.vbCrLf;
			return code;

		}
		public object creaFriends()
		{
			string code = string.Empty;
			code += "Friend WithEvents CtlABCMaestro As Sist" + this.neimespeis + ".ctlABCMaestro" + Constants.vbCrLf;
			code += "Friend WithEvents CtlbuscaReg As Sist" + this.neimespeis + ".ctlbuscaReg" + Constants.vbCrLf;
			code += "Friend WithEvents pnlDatos As System.Windows.Forms.Panel" + Constants.vbCrLf;
			code += "Friend WithEvents lblFalta As System.Windows.Forms.Label" + Constants.vbCrLf;
			foreach (Propiedad atr in this.clace.Campos) {
				string tipoCtl = atr.tipoControl;
				string nonCtl = atr.nombreControl;
				code += "Friend WithEvents lbl" + atr.Nombre + " As System.Windows.Forms.Label" + Constants.vbCrLf;

				if (atr.Nombre.Substring(atr.Nombre.Length - 3, 3).Equals("_id") & !this.clace.llave.Equals(atr.Nombre)) {
					tipoCtl = "cmb";
					nonCtl = "ComboBox";
				}
				code += "Friend WithEvents " + tipoCtl + atr.Nombre + " As System.Windows.Forms." + nonCtl + Constants.vbCrLf;
			}

			return code;
		}
		public string creaControles(Propiedad atr)
		{
			string code = string.Empty;
			code += this.creaEtiqueta(atr);
			code += this.creaCapturador(atr);
			return code;
		}

		public string creaEtiqueta(Propiedad atr)
		{
			string code = string.Empty;
			code += "'" + Constants.vbCrLf;
			code += "'lbl" + atr.titulo + Constants.vbCrLf;
			code += "'" + Constants.vbCrLf;
			code += "Me.lbl" + atr.titulo + ".AutoSize = True" + Constants.vbCrLf;
			code += "Me.lbl" + atr.titulo + ".Font = ";
			code += "New System.Drawing.Font(\"Microsoft Sans Serif\", 9.75!, ";
			code += "System.Drawing.FontStyle.Bold, ";
			code += "System.Drawing.GraphicsUnit.Point, CType(0, Byte))" + Constants.vbCrLf;
			code += "Me.lbl" + atr.titulo + ".Location = New System.Drawing.Point(20, " + (25 * this.tabIndex) + 5 + ")" + Constants.vbCrLf;
			code += "Me.lbl" + atr.titulo + ".Name = \"lbl" + atr.titulo + "\"" + Constants.vbCrLf;
			code += "Me.lbl" + atr.titulo + ".Size = New System.Drawing.Size(" + atr.Nombre.Length * 10 + ", 16)" + Constants.vbCrLf;
			code += "Me.lbl" + atr.titulo + ".TabIndex = 0" + Constants.vbCrLf;
			code += "Me.lbl" + atr.titulo + ".Text = \"" + atr.titulo + "\"" + Constants.vbCrLf;
			return code;
		}
		public string creaCapturador(Propiedad atr)
		{
			string code = string.Empty;
			string tipoCtl = atr.tipoControl;

			if (atr.Nombre.Substring(atr.Nombre.Length - 3, 3).Equals("_id") & !this.clace.llave.Equals(atr.Nombre)) {
				tipoCtl = "cmb";

			}
			code += "'" + Constants.vbCrLf;
			code += "'" + tipoCtl + atr.titulo + Constants.vbCrLf;
			code += "'" + Constants.vbCrLf;
			code += "Me." + tipoCtl + atr.titulo + ".Enabled = False" + Constants.vbCrLf;
			code += "Me." + tipoCtl + atr.titulo + ".Location =  New System.Drawing.Point(150, " + (25 * this.tabIndex) + 5 + ")" + Constants.vbCrLf;
			code += "Me." + tipoCtl + atr.titulo + ".Name = \"" + tipoCtl + atr.titulo + "\"" + Constants.vbCrLf;
			if (atr.Nombre.Equals(this.clace.llave)) {
				code += "Me." + tipoCtl + atr.titulo + ".ReadOnly = True" + Constants.vbCrLf;
			}
			code += "Me." + tipoCtl + atr.titulo + ".Size = New System.Drawing.Size(100, 20)" + Constants.vbCrLf;
			code += "Me." + tipoCtl + atr.titulo + ".TabIndex = " + this.tabIndex + Constants.vbCrLf;
			this.tabIndex += 1;
			return code;
		}
		public string creaForma()
		{
			string code = string.Empty;
			code += "'" + Constants.vbCrLf;
			code += "'frm" + this.clace.NombreTabla + Constants.vbCrLf;
			code += "'" + Constants.vbCrLf;
			code += "Me.AutoScaleDimensions = New System.Drawing.SizeF(6.0!, 13.0!)" + Constants.vbCrLf;
			code += "Me.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font" + Constants.vbCrLf;
			code += "Me.ClientSize = New System.Drawing.Size(300," + (this.clace.Campos.Count * 30) + 100 + ")" + Constants.vbCrLf;
			code += "Me.Controls.Add(Me.pnlDatos)" + Constants.vbCrLf;
			code += "Me.Controls.Add(Me.CtlbuscaReg)" + Constants.vbCrLf;
			code += "Me.Controls.Add(Me.CtlABCMaestro)" + Constants.vbCrLf;
			code += "Me.Name = \"frm" + this.clace.NombreTabla + "\"" + Constants.vbCrLf;
			code += "Me.Text = \"" + this.clace.NombreTabla + "\"" + Constants.vbCrLf;
			code += "Me.pnlDatos.ResumeLayout(False)" + Constants.vbCrLf;
			code += "Me.pnlDatos.PerformLayout()" + Constants.vbCrLf;
			code += "Me.ResumeLayout(False)" + Constants.vbCrLf;
			return code;
		}
		public string creaBusqueda()
		{
			string code = string.Empty;
			code += "'" + Constants.vbCrLf;
			code += "'CtlbuscaReg" + Constants.vbCrLf;
			code += "'" + Constants.vbCrLf;
			code += "Me.CtlbuscaReg.Location = New System.Drawing.Point(0, 0)" + Constants.vbCrLf;
			code += "Me.CtlbuscaReg.Name = \"CtlbuscaReg\"" + Constants.vbCrLf;
			code += "Me.CtlbuscaReg.Size = New System.Drawing.Size(516, 25)" + Constants.vbCrLf;
			code += "Me.CtlbuscaReg.TabIndex = 1" + Constants.vbCrLf;
			return code;
		}
		public string creaMaestro()
		{
			string code = string.Empty;
			code += "'" + Constants.vbCrLf;
			code += "'CtlABCMaestro" + Constants.vbCrLf;
			code += "'" + Constants.vbCrLf;
			code += "Me.CtlABCMaestro.Estado = 0" + Constants.vbCrLf;
			code += "Me.CtlABCMaestro.Location = New System.Drawing.Point(0, 27)" + Constants.vbCrLf;
			code += "Me.CtlABCMaestro.Name = \"CtlABCMaestro\"" + Constants.vbCrLf;
			code += "Me.CtlABCMaestro.Size = New System.Drawing.Size(516, 25)" + Constants.vbCrLf;
			code += "Me.CtlABCMaestro.TabIndex = 0" + Constants.vbCrLf;
			return code;
		}






	}
}
